commit bb244d0772b60b65693b77158ed589eee71ff5a4 (refs/changes/14/1714/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-04-04T15:00:56+00:00 (2 years, 6 months ago)
    
    TOOLS-1727 sdcadm create cloudapi does not give it an external nic

diff --git a/lib/cli/do_update_other.js b/lib/cli/do_update_other.js
index 7e33b65..84e542b 100644
--- a/lib/cli/do_update_other.js
+++ b/lib/cli/do_update_other.js
@@ -752,7 +752,30 @@ function do_update_other(subcmd, opts, args, cb) {
             }, next);
         },
 
-        steps.agentServicesEnsureCreated
+        steps.agentServicesEnsureCreated,
+
+        function updateCloudapiCfg(ctx, next) {
+            var cloudapiSvc = ctx.svcs.filter(function (svc) {
+                return svc.name === 'cloudapi';
+            })[0];
+            if (cloudapiSvc.params.networks.length > 1) {
+                next();
+                return;
+            }
+            progress('Add "external" network parameter to "CloudAPI" service');
+            updateService(cloudapiSvc.uuid, { params: {
+                networks: [
+                    { name: 'admin' },
+                    { name: 'external', primary: true}
+                ]
+            } }, function (cloudapiErr) {
+                if (cloudapiErr) {
+                    next(new errors.SDCClientError(cloudapiErr, 'sapi'));
+                    return;
+                }
+                next();
+            });
+        }
 
     ]}, function (err) {
         if (!history) {
